test/ctia/test_helpers/aggregate.clj

```diff
@@ -149,19 +154,29 @@
        (map (fn [[k v]]
               {:key (str k) :value v}))))
 
+(defn- vals->date-vals [from-str to-str values]
+  (let [from  (tf/parse from-str)
+        to    (tf/parse to-str)
+        parse (fn [d] (cond-> d
+                        (inst? d) format-date-time
+                        d tf/parse))]
+    (->> values
+         (map parse)
+         (filter #(and (time/within? from to %)
+                       (time/before? % to))))))
+
 (defn- test-histogram
   "test one field histogram, examples are already created"
   [examples entity field granularity]
   (testing (format "histogram %s %s" entity field)
     (let [parsed (parse-field field)
-          values (keep #(get-in % parsed) examples)
+          values (->>
+                  examples
+                  (keep #(es-get-in % parsed))
+                  flatten)
           from-str "2020-03-01T00:00:00.000Z"
           to-str "2020-10-01T00:00:00.000Z"
-          from (tf/parse from-str)
-          to (tf/parse to-str)
-          date-values (filter #(and (time/within? from to %)
-                                    (time/before? % to))
-                              (map tf/parse values))
+          date-values (vals->date-vals from-str to-str values)
```

Comment:
Since this test helper is tricky, I would add check that `date-values` is not empty:
``` clojure
(assert (ses date-values))
```